src/pages/Balances/Balances.tsx

```diff
@@ -0,0 +1,38 @@
+import { isArweaveTransactionID, isEthAddress } from '@src/utils';
+import { ArweaveTransactionID } from '@src/utils/ArweaveTransactionId';
+import { useMemo } from 'react';
+import { useParams } from 'react-router-dom';
+import BalancesHeader from './BalancesHeader';
+import Banner from './Banner';
+import VaultsTable from './VaultsTable';
+
+const Balances = () => {
+  const params = useParams();
+  const walletAddress = params?.walletAddress;
+
+  const walletAddressData = useMemo(() => {
+    return walletAddress == undefined
```

Comment:
can simplify to !isAoAddress(walletAddress) ? undefined : ... eth/arweave logic

Or, might be worth have a util for new AoAddress() which handles this logic

Reply:
Could do that here since isAOAddress checks against ArweaveTransactionId and not string. It's a bit quirky and may try to get rid of ArweaveTransactionId in the future and replace with type guard that uses string.